packages/xdl/src/start/ExpoUpdatesManifestHandler.ts

```diff
@@ -29,14 +36,80 @@ function getPlatformFromRequest(req: express.Request | http.IncomingMessage): st
   return String(platform);
 }
 
+async function shouldUseAnonymousManifestAsync(): Promise<boolean> {
+  const currentSession = await UserManager.getSessionAsync();
+  return !currentSession || ConnectionStatus.isOffline();
+}
+
+async function getScopeKeyAndProjectIdForExpoConfigAsync(
+  expoConfig: ExpoConfig
+): Promise<{
+  scopeKey: string;
+  projectId: string | null;
+}> {
+  const shouldUseAnonymousManifest = await shouldUseAnonymousManifestAsync();
+  if (shouldUseAnonymousManifest) {
+    const userAnonymousIdentifier = await UserSettings.getAnonymousIdentifierAsync();
+    return {
+      scopeKey: `@${ANONYMOUS_USERNAME}/${expoConfig.slug}-${userAnonymousIdentifier}`,
```

Comment:
I suspect we'll want to change this in the future not to follow the `@account/project` format and possibly to move away from using the slug (if we drop that field). But I don't think that should stop this PR and there shouldn't be high expectations for scope keys to be to be consistent for anonymous projects, especially if the anonymous identifier can change from computer to computer.

Reply:
yep, agreed. The key here though is that i _think_ there's code in the client that looks explicitly for this: https://github.com/expo/expo/blob/master/android/expoview/src/main/java/host/exp/exponent/ExponentManifest.kt#L144

It actually seems like any `id` that starts with `@anonymous/` is considered verified, so that is independent of scopeKey. Hmm.